### DISCUSSION TITLE
Is there a way to include inline js?

### DISCUSSION THREAD

**Discussion by j2deme** [marp-team/marp#483] [Q&A] [answered]
Let's say I want to include a Google map with some pins, lines o polygons on it (not an image) or maybe a D3js graph to show show charts, or an example for graph theory, or some sort of dynamic data structure visualization...

I may also see a use case to get dynamic data from an API when the most updated value is needed 🤔 or when you want to show info as the number of repos or stars, or maybe the last tweet or an specific tweet (I don't now if they are still called _tweets_ 😅) from a user.

I haven't tried yet, but, is there a way to use inline or external js file to get the proposed result?

**Comment by yhatt** [comment, marked as answer]
By the CommonMark spec, Markdown actually allows to contain the `<script>` element, to use both of inline and external scripts.
https://spec.commonmark.org/0.30/#example-170

For security reasons, Marp tools are disallowed containing the most of HTML tags by default, including `<script>`. But you can allow using HTML tags by the preferences for each tools.

- [Marp CLI](https://github.com/marp-team/marp-cli): Pass `--html` option to allow HTML elements.
- [Marp for VS Code](https://github.com/marp-team/marp-vscode): Turn on `Markdown > Marp: Enable HTML` from the extension setting. You may also have to relax [the security setting of VS Code's Markdown preview](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview-security) to run the script.

#250 might be helpful if you want to use the inline script.

**Reply by j2deme** [reply]
Thank you for taking the time to answer these kinds of questions, I'll try a couple of examples for the given proposed use cases.

I totally ❤️ Marp, it helps a lot in my daily work as a IT professor.
